<Type Name="InvalidPipelineStoreException" FullName="System.AddIn.Hosting.InvalidPipelineStoreException">
  <TypeSignature Language="C#" Value="public class InvalidPipelineStoreException : Exception" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit InvalidPipelineStoreException extends System.Exception" />
  <TypeSignature Language="DocId" Value="T:System.AddIn.Hosting.InvalidPipelineStoreException" />
  <TypeSignature Language="VB.NET" Value="Public Class InvalidPipelineStoreException&#xA;Inherits Exception" />
  <TypeSignature Language="C++ CLI" Value="public ref class InvalidPipelineStoreException : Exception" />
  <AssemblyInfo>
    <AssemblyName>System.AddIn</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
  </Base>
  <Interfaces />
  <Docs>
    <summary><span data-ttu-id="a3264-101">La excepción que se produce cuando no se encuentra un directorio y el usuario no tiene permiso de acceso a la ruta raíz de la canalización ni a la ruta de un complemento.</span><span class="sxs-lookup"><span data-stu-id="a3264-101">The exception that is thrown when a directory is not found and the user does not have permission to access the pipeline root path or an add-in path.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="a3264-102">A diferencia de <xref:System.IO.DirectoryNotFoundException>, esta excepción no proporciona ninguna información sobre la ruta de acceso.</span><span class="sxs-lookup"><span data-stu-id="a3264-102">Unlike <xref:System.IO.DirectoryNotFoundException>, this exception does not provide any path information.</span></span> <span data-ttu-id="a3264-103">Esto evita que un usuario malintencionado pueda especificar deliberadamente rutas de acceso erróneas y utilizar la información que devuelve la excepción para determinar las rutas de acceso correctas.</span><span class="sxs-lookup"><span data-stu-id="a3264-103">This prevents a malicious user from deliberately specifying erroneous paths and using the information returned by the exception to determine actual paths.</span></span>  
  
 <span data-ttu-id="a3264-104">Esta excepción se produce por los siguientes métodos de detección que compilan y actualizan el almacén de información de complemento y la canalización: <xref:System.AddIn.Hosting.AddInStore.FindAddIns%2A>,<xref:System.AddIn.Hosting.AddInStore.Rebuild%2A>, <xref:System.AddIn.Hosting.AddInStore.RebuildAddIns%2A>, <xref:System.AddIn.Hosting.AddInStore.Update%2A>, y <xref:System.AddIn.Hosting.AddInStore.UpdateAddIns%2A>.</span><span class="sxs-lookup"><span data-stu-id="a3264-104">This exception is thrown by the following discovery methods that build and update the store of add-in and pipeline information: <xref:System.AddIn.Hosting.AddInStore.FindAddIns%2A>,<xref:System.AddIn.Hosting.AddInStore.Rebuild%2A>, <xref:System.AddIn.Hosting.AddInStore.RebuildAddIns%2A>, <xref:System.AddIn.Hosting.AddInStore.Update%2A>, and <xref:System.AddIn.Hosting.AddInStore.UpdateAddIns%2A>.</span></span>  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="a3264-105">Inicializa una nueva instancia de la clase <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" />.</span><span class="sxs-lookup"><span data-stu-id="a3264-105">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public InvalidPipelineStoreException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AddIn.Hosting.InvalidPipelineStoreException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; InvalidPipelineStoreException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="a3264-106">Inicializa una nueva instancia de la clase <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" />.</span><span class="sxs-lookup"><span data-stu-id="a3264-106">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public InvalidPipelineStoreException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AddIn.Hosting.InvalidPipelineStoreException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; InvalidPipelineStoreException(System::String ^ message);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="a3264-107">Descripción del error.</span><span class="sxs-lookup"><span data-stu-id="a3264-107">A description of the error.</span></span></param>
        <summary><span data-ttu-id="a3264-108">Inicializa una nueva instancia de la clase <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> con el mensaje especificado.</span><span class="sxs-lookup"><span data-stu-id="a3264-108">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class with the specified message.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected InvalidPipelineStoreException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AddIn.Hosting.InvalidPipelineStoreException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; InvalidPipelineStoreException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="a3264-109">Datos del objeto serializado.</span><span class="sxs-lookup"><span data-stu-id="a3264-109">The serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="a3264-110">Información contextual sobre los datos del objeto de origen o destino.</span><span class="sxs-lookup"><span data-stu-id="a3264-110">The contextual information about the source or destination object data.</span></span></param>
        <summary><span data-ttu-id="a3264-111">Inicializa una nueva instancia de la clase <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> con la información de serialización y del contexto de transmisión por secuencias especificada.</span><span class="sxs-lookup"><span data-stu-id="a3264-111">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class with serialization and streaming context information.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="a3264-112">Se llama a este constructor durante la deserialización para reconstituir un objeto de excepción transmitido en una secuencia.</span><span class="sxs-lookup"><span data-stu-id="a3264-112">This constructor is called during deserialization to reconstitute an exception object that is transmitted over a stream.</span></span> <span data-ttu-id="a3264-113">Para obtener más información, consulte [serialización XML y SOAP](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="a3264-113">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public InvalidPipelineStoreException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AddIn.Hosting.InvalidPipelineStoreException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; InvalidPipelineStoreException(System::String ^ message, Exception ^ innerException);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.AddIn</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="a3264-114">Descripción del error.</span><span class="sxs-lookup"><span data-stu-id="a3264-114">A description of the error.</span></span></param>
        <param name="innerException"><span data-ttu-id="a3264-115">Excepción que es la causa de la excepción actual.</span><span class="sxs-lookup"><span data-stu-id="a3264-115">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="a3264-116">Si el parámetro <c>innerException</c> no es una referencia nula, la excepción se produce en un bloque catch que controla la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="a3264-116">If the <c>innerException</c> parameter is not a null reference, the current exception is raised in a catch block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="a3264-117">Inicializa una nueva instancia de la clase <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> con el mensaje de error especificado y una referencia a la excepción interna que representa la causa de esta excepción.</span><span class="sxs-lookup"><span data-stu-id="a3264-117">Initializes a new instance of the <see cref="T:System.AddIn.Hosting.InvalidPipelineStoreException" /> class with the specified message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="a3264-118">Una excepción que se produce como consecuencia directa de una excepción anterior debe incluir una referencia a esta última en la propiedad <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="a3264-118">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="a3264-119">El <xref:System.Exception.InnerException%2A> propiedad devuelve el mismo valor que se pasa al constructor, o una referencia nula si el <xref:System.Exception.InnerException%2A> propiedad no proporciona el valor de la excepción interna al constructor.</span><span class="sxs-lookup"><span data-stu-id="a3264-119">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed to the constructor, or a null reference if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="a3264-120">En la siguiente tabla se muestran los valores iniciales de las propiedades de una instancia de la clase <xref:System.AddIn.Hosting.AddInSegmentDirectoryNotFoundException>.</span><span class="sxs-lookup"><span data-stu-id="a3264-120">The following table shows the initial property values for an instance of the <xref:System.AddIn.Hosting.AddInSegmentDirectoryNotFoundException> class.</span></span>  
  
|<span data-ttu-id="a3264-121">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="a3264-121">Property</span></span>|<span data-ttu-id="a3264-122">Valor</span><span class="sxs-lookup"><span data-stu-id="a3264-122">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="a3264-123">Referencia a la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="a3264-123">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="a3264-124">La cadena del mensaje de error localizado.</span><span class="sxs-lookup"><span data-stu-id="a3264-124">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>